mlrun/serving/states.py

```diff
@@ -388,6 +366,40 @@ def init_object(self, context, namespace, mode="sync", reset=False, **extra_kwar
         if mode != "skip":
             self._post_init(mode)
 
+    def get_full_class_args(self, namespace, class_object, **extra_kwargs):
+        class_args = {}
+        for key, arg in self.class_args.items():
+            if key.startswith(callable_prefix):
+                class_args[key[1:]] = get_function(arg, namespace)
+            else:
+                class_args[key] = arg
+        class_args.update(extra_kwargs)
+
+        # add common args (name, context, ..) only if target class can accept them
+        argspec = getfullargspec(class_object)
+        for key in ["name", "context", "input_path", "result_path", "full_event"]:
+            if argspec.varkw or key in argspec.args:
+                class_args[key] = getattr(self, key)
+        if argspec.varkw or "graph_step" in argspec.args:
+            class_args["graph_step"] = self
+        return class_args
+
+    def get_or_create_class_object(self, namespace):
```

Comment:
I find it a bit confusing, because you don't really creates anything .  
```suggestion
    def get_step_class_object(self, namespace):
```

Reply:
great idea!